agent/src/runtime_connectors/podman_cli.rs

```diff
@@ -91,9 +105,15 @@ impl TimedPodmanPsResult {
     }
 
     async fn new_inner() -> PodmanPsCache {
+        let mut res = PodmanCli::list_states_internal().await;
+        if res.is_err() {
+            // This is a workaround for the known issue in podman (podman ps sometimes fails).
+            log::trace!("'podman ps' has returned error - let's retry it.");
+            res = PodmanCli::list_states_internal().await;
+        }
```

Comment:
Is this sufficient? Could it be that the second try fails again with the same error?
Maybe we can try to retry a couple (e.g., 3) of times in a loop.

Reply:
That is good question. I was considering that. I was testing this implementation on VM and on the target. My experience says that the error does not happen more than once. Since this is just an workaround, I would suggest to keep the implementation simple (as it is here).

Reply:
Point taken.